### PR TITLE
feat: [CI-5775]: Docker addon file

### PR DESCRIPTION
**Pull request by HenAmar**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41449)

### CODE REVIEW

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41449#-:-NKSiW-S7TAY2ATvnOqY:bnfp4nl)* status: Some of the checks are too old.

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 1]
```diff
@@ -25,8 +25,10 @@ RUN microdnf update \
 COPY ci-addon_/ci-addon /usr/local/bin/ci-addon-linux
 COPY java-agent.jar /usr/local/bin/
 COPY split_tests /usr/local/bin/
+COPY tmate-2.4.0-static-linux-arm64v8/tmate /usr/local/bin/
```
Comment (harshjain12) [inline product/ci/addon/docker/Dockerfile.linux.arm64:28]: tmate is part of image in arm64 ?
why we have to download in amd ?

Comment (HenAmar) [reply product/ci/addon/docker/Dockerfile.linux.arm64:28]: In arm i made the download process in the addon build pipeline. The download/installation/extraction wasn't working well when building the amd image.

**Review by harshjain12 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41449)* status: Some of the checks are too old.

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41449#-:-NKSp9FF0p-_zPXiFbXB:bnfp4nl)* status: Some of the checks are too old.

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41449#-:-NKSpuWt8dBZOdhFWUXy:bnfp4nl)* status: Some of the checks are too old.

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41449#-:-NKSq4h-4gK0Q-p9Fmxk:bnfp4nl)* status: Some of the checks are too old.

**Pull request merged**